File: BOJ/Graph/Silver/1260/1260.swift
var ans = ""
let input = readLine()!.split(separator : " ").map{Int(String($0))!}
let n = input[0]
let m = input[1]
let v = input[2]
var visit = Array(repeating : false, count : n+1)
var nodes = Array(repeating : [Int](), count : n+1)
for _ in 0..<m {
    let vertexs = readLine()!.split(separator: " ").map{Int(String($0))!}
    nodes[vertexs[0]].append(vertexs[1])
    nodes[vertexs[1]].append(vertexs[0])
}
for i in 1...n {nodes[i].sort()} 
dfs(v)
print(ans)
visit.indices.forEach{visit[$0] = false}
bfs()

func dfs(_ cur : Int) {
    visit[cur] = true
    ans += "\(cur) "
    for next in nodes[cur] {
        if visit[next] {continue}
        dfs(next)
    }
}
func bfs() {
    var str = ""
    var q : [Int] = []
    q.append(v)
    visit[v] = true
    str += "\(v) "
    while q.count > 0 {
        let current = q.removeFirst()
        for next in nodes[current] {
            if visit[next] {continue}
            visit[next] = true
            q.append(next)
            str += "\(next) "
        }
    }
    print(str)
}

File: BOJ/Graph/Silver/1260/1260.py
import sys
from collections import deque
ans = ""
def dfs(cur) :
    global ans
    visit[cur] = True
    ans += f"{cur} "
    for next in nodes[cur] :
        if visit[next] : continue
        dfs(next)
def bfs(ans) :
    q = deque()
    q.append(v)
    visit[v] = True
    ans += f"{v} "
    while len(q) > 0 :
        current = q.popleft()
        for next in nodes[current] :
            if visit[next] : continue
            visit[next] = True
            q.append(next)
            ans += f"{next} "
    print(ans)
n,m,v = list(map(int,sys.stdin.readline().split()))
visit = [False] * (n+1)
nodes = [[] for _ in range(n+1)]
for _ in range(m) :
    _from,_to = list(map(int,sys.stdin.readline().split()))
    nodes[_from].append(_to)
    nodes[_to].append(_from)
for node in nodes : node.sort()
dfs(v)
print(ans)
visit = list(map(lambda x : False, visit))
bfs("")